### PR TITLE
STAT-136 fix folder removal when failing

### PR DESCRIPTION
**Pull request by Andilun**
Creation of project now happens inside context managed temporary folder. If any function fails after the project folder is created, the folder is deleted.

When a project is created successfully it is moved to $HOME or user specified path.

Build and create now uses cwd instead of $HOME as default build path. (https://statistics-norway.atlassian.net/browse/STAT-176)

### CODE REVIEW

File: tests/test_app.py [hunk 1]
```diff
@@ -313,3 +314,33 @@ def test_get_kernels_dict(mock_run: Mock) -> None:
...
+        f"{PKG}.create_project_from_template", return_value=None
+    ) as mock_template:
+        with patch(f"{PKG}.make_and_init_git_repo", return_value=None):
+            with patch(f"{PKG}.poetry_install", return_value=None):
+                with patch(
+                    f"{PKG}.install_ipykernel", return_value=None
+                ) as mock_kernel:
+                    with patch(f"{PKG}.copytree", return_value=None) as mock_copy:
+                        mock_template.side_effect = ValueError("template failing")
+                        with pytest.raises(ValueError):
```
Comment (mmwinther) [inline tests/test_app.py:334]: There are three individual test cases here, we should split them into individual tests.

File: tests/test_app.py [hunk 2]
```diff
@@ -313,3 +314,33 @@ def test_get_kernels_dict(mock_run: Mock) -> None:
...
+    """
+    with patch(
+        f"{PKG}.create_project_from_template", return_value=None
+    ) as mock_template:
+        with patch(f"{PKG}.make_and_init_git_repo", return_value=None):
+            with patch(f"{PKG}.poetry_install", return_value=None):
+                with patch(
+                    f"{PKG}.install_ipykernel", return_value=None
+                ) as mock_kernel:
+                    with patch(f"{PKG}.copytree", return_value=None) as mock_copy:
```
Comment (mmwinther) [inline tests/test_app.py:332]: These mocks could be put into fixtures to make the tests cleaner and make it easier to write multiple smaller tests

**Review by mmwinther — COMMENTED** [review]
Looks good, the only thing I think needs improvement is the test 😄

File: tests/test_app.py (hunk 1 above)
Comment (Andilun) [reply tests/test_app.py]: Tests are now split up.

File: tests/test_app.py (hunk 2 above)
Comment (Andilun) [reply tests/test_app.py]: Set up a test class with mocks, should be easier to add more tests for the create function now.

**Review by mmwinther — APPROVED** [review]
Nice work 👍

**Review by mmwinther — APPROVED** [review]

**Pull request merged**